warehouse/query-core/src/main/java/datawave/query/function/MatchingFieldGroup.java

```diff
@@ -0,0 +1,26 @@
+package datawave.query.function;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class MatchingFieldGroup {
+    private final Set<String> matchingFieldSet;
+    private final Set<String> matchingValues;
```

Comment:
Do we need to organize the values by field?  Is value matching across fields ok?  Can we add a test showcasing that if that was the intent?

Reply:
The matching values contain values for one of the fields that is a hit in the query or of the same group as a hit in the query.  So this is simply keeping track of values that we would match to other fields in this fields set.  Perhaps I need to rename it?

Reply:
I see what you mean.  Looking at this again,  you aren't using the `matchingFieldSet` variable in this class.  The `containsField` method is unused as well.  Is there a good reason to keep the field set if we're not using it?

Reply:
This comment still stands I think.  Why do we need matchingFieldSet?